cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/cloudstorage/ImportCompletionCoordinator.java

```diff
@@ -42,31 +43,36 @@
 
 import o.a.c.sidecar.client.shaded.common.request.data.CreateSliceRequestPayload;
 import org.apache.cassandra.sidecar.client.SidecarInstance;
-import org.apache.cassandra.spark.bulkwriter.blobupload.BlobDataTransferApi;
-import org.apache.cassandra.spark.bulkwriter.blobupload.BlobStreamResult;
-import org.apache.cassandra.spark.bulkwriter.blobupload.CreatedRestoreSlice;
+import org.apache.cassandra.spark.bulkwriter.BulkWriteValidator;
+import org.apache.cassandra.spark.bulkwriter.BulkWriterContext;
+import org.apache.cassandra.spark.bulkwriter.CancelJobEvent;
+import org.apache.cassandra.spark.bulkwriter.ClusterInfo;
+import org.apache.cassandra.spark.bulkwriter.JobInfo;
+import org.apache.cassandra.spark.bulkwriter.RingInstance;
 import org.apache.cassandra.spark.data.ReplicationFactor;
+import org.apache.cassandra.spark.exception.ImportFailedException;
 import org.apache.cassandra.spark.transports.storage.extensions.StorageTransportExtension;
 import org.apache.cassandra.util.ThreadUtil;
 
 import static org.apache.cassandra.clients.Sidecar.toSidecarInstance;
-import static org.apache.cassandra.spark.bulkwriter.blobupload.CreatedRestoreSlice.ConsistencyLevelCheckResult.NOT_SATISFIED;
-import static org.apache.cassandra.spark.bulkwriter.blobupload.CreatedRestoreSlice.ConsistencyLevelCheckResult.SATISFIED;
+import static org.apache.cassandra.spark.bulkwriter.cloudstorage.CreatedRestoreSlice.ConsistencyLevelCheckResult.NOT_SATISFIED;
+import static org.apache.cassandra.spark.bulkwriter.cloudstorage.CreatedRestoreSlice.ConsistencyLevelCheckResult.SATISFIED;
 
-public final class ImportCompletionCoordinator
+public final class ImportCompletionCoordinator implements ImportCoordinator
```

Comment:
Does it make sense to call this class something like `TimeBoundedImportCoordinator` ?

Reply:
Maybe the javadoc should explain what are the terminal conditions for the coordinated import?